Text/LaTeX/Base/Render.hs

```diff
@@ -161,17 +161,17 @@ instance Render LaTeX where
 
   render = renderDefault
 
-instance Render TeXArg where
- renderBuilder (FixArg l) = "{" <> renderBuilder l <> "}"
- renderBuilder (OptArg l) = "[" <> renderBuilder l <> "]"
- renderBuilder (MOptArg []) = mempty
- renderBuilder (MOptArg ls) = "[" <> renderCommasBuilder ls <> "]"
- renderBuilder (SymArg l) = "<" <> renderBuilder l <> ">"
- renderBuilder (MSymArg []) = mempty
- renderBuilder (MSymArg ls) = "<" <> renderCommasBuilder ls <> ">"
- renderBuilder (ParArg l) = "(" <> renderBuilder l <> ")"
- renderBuilder (MParArg []) = mempty
- renderBuilder (MParArg ls) = "(" <> renderCommasBuilder ls <> ")"
+instance (Show a) => Render (TeXArg a) where
```

Comment:
Why is `Show` needed here?

Reply:
I asked myself the same question, it comes as a superclass of `Render`, which is delcared as:

```haskell
class (Show a) => Render a ...
```

Reply:
Interesting. Maybe that's something to overthink independently... I suspect this might be an artifact from the time when `Show` was a superclass of `Num`, or something silly like that.

Reply:
> Interesting. Maybe that's something to overthink independently... I suspect this might be an artifact from the time when `Show` was a superclass of `Num`, or something silly like that.

It was in order to have a default method that uses `show`. Maybe not the best idea in restrospect...

Reply:
@Daniel-Diaz `-XDefaultSignatures`?

Reply:
> @Daniel-Diaz `-XDefaultSignatures`?

Yes. It could be changed to use that. But currently I think `show` is the wrong function for this use case, as `show` is supposed to generate Haskell code that can be used to produce the value back. However, removing the `Show` superclass now might break some code that uses the default implementation. So default signatures is probably the best solution at the moment.

Reply:
It might still break some code that relies on the superclass, but such code perhaps _should_ be broken.